atc/config_diff.go

```diff
@@ -53,6 +58,24 @@ func (diff Diff) Render(to io.Writer, label string) {
 	}
 }
 
+func (diff DisplayDiff) Render(to io.Writer) {
+	label := "display configuration"
+	if diff.Before != nil && diff.After != nil {
+		fmt.Fprintf(to, ansi.Color("%s has changed:", "yellow")+"\n", label)
+		payloadA, _ := yaml.Marshal(diff.Before)
+		payloadB, _ := yaml.Marshal(diff.After)
+		renderDiff(to, string(payloadA), string(payloadB))
```

Comment:
We probably shouldn't be rendering diffs if either Before or After is a base64 encoded image. A short summary or message like `((base64 encoded image omitted))` is a lot more meaningful than the actual value.

Reply:
That's a good point (it is kinda ugly). I'm split as to whether to allow base64 encoded images as they have the benefit of not needing to egress to some random location as they are entirely contained within the pipeline config and the overhead of transferring that data. I think maybe just allowing links only is the simplest case and users who are running in an egress-controlled environment will either not be able to use the memes or will have to host them themselves (it is easy enough to just get the web node to host them on `/public/images` anyway.

Reply:
I don't have a strong opinion either way. I guess that the teams that actually need this feature (i.e. differentiate between pipelines) are more likely to be in an enterprise environment. Which probably means that they have access to a s3-like bucket to serve the images.